### PR TITLE
Add redux for state management

### PR DESCRIPTION
**Pull request by dnivra26**
https://github.com/vimark1/typist/issues/49

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 import firebase from 'firebase/app';
+import { Provider } from 'react-redux';
 import 'firebase/auth';
 
 import './index.css';
 
 import App from './App';
 import * as serviceWorker from './serviceWorker';
 import firebaseCred from './firebase-cred.json';
+import configureStore from './store/configureStore';
```
Comment (vimark1) [inline src/index.js]: Missing file

Comment (dnivra26) [reply src/index.js]: oh my bad. forgot the new files :|
pushed now

Comment (dnivra26) [reply src/index.js]: and i think the builds are also passing now

Comment (vimark1) [reply src/index.js]: great, thanks @dnivra26 - will you be sending more PRs regarding redux?

Comment (dnivra26) [reply src/index.js]: this PR is just adding redux to the project. If you can tell how we are planning to use redux, like what to store in the redux store, should we use middlewares like thunk, etc.. i can help

Comment (vimark1) [reply src/index.js]: @dnivra26 I am not sure tbh, I don't have experience working with redux, so I am keen to know your opinion, ie what you think is best.

Comment (dnivra26) [reply src/index.js]: cool. no problem. ill go through the app and the code in detail tomorrow and come up with my recommendations

Comment (vimark1) [reply src/index.js]: Awesome, will you push the changes to the same branch?

Comment (dnivra26) [reply src/index.js]: probably as part of different PRs

Comment (dnivra26) [reply src/index.js]: sure

**Comment by vimark1** [comment]
@dnivra26 I might merge this as it is because we can still get other's contribution using redux.  Please open a new PR for more redux implementation

**Pull request merged**